bin/kano-updater-recovery

```diff
@@ -0,0 +1,351 @@
+#!/usr/bin/env python
+
+# kano-updater-recovery
+#
+# Copyright (C) 2018 Kano Computing Ltd.
+# License: http://www.gnu.org/licenses/gpl-2.0.txt GNU GPL v2
+#
+# Recovery flow for the updater.
+# Needs to be run with sudo.
+
+
+MINUTE = 60  # seconds
+HOUR = 60 * MINUTE
+
+SYSTEM_PROGRESS_TIMEOUT = 2 * HOUR
+PERMITTED_WIFI_RETRIES = 3
+RECOVERY_STATUS_PATH = '/var/cache/kano-updater/recovery.txt'
+
+
+class RC(object):
+    """
+    Return code constants
+    """
+
+    SUCCESS = 0
+    UF1 = 1  # dpkg couldn't resolve problems (shouldn't be called from here)
+    UF2 = 2  # Problem launching the updater
+    UF3 = 3  # WiFi connection problems
+    UF4 = 4  # WiFi retries exceeded
+    UF5 = 5  # Progress update timeout
+    UF6 = 6  # System timeout
+
+    @staticmethod
+    def rc_to_str(rc_val):
+        """
+        Returns an RC string for the supplied RC value corresponding to the
+        variable name as defined above.
+
+        For example, RC.rc_to_str(RC.UF1) returns 'UF1'.
+
+        Look through the RC object, ignoring all private members, checking for
+        equality, returning the string of the property found.
+        """
+
+        for attr_name in filter(lambda x: not x.startswith('_'), dir(RC)):
+            attr = getattr(RC, attr_name)
+            if attr == rc_val:
+                return attr_name
+
+        # Not found but return gracefully
+        return ''
+
+
+def _stop_systemd_service(service):
+    """
+    Helper function to stop SystemD services
+
+    Args:
+        service (str): Name of the SystemD service to kill
+    """
+
+    import os
+    os.system('systemctl stop {}'.format(service))
+
+
+def _start_systemd_service(service):
+    """
+    Helper function to launch SystemD services
+
+    Args:
+        service (str): Name of the SystemD service to launch
+    """
+
+    import os
+    os.system('systemctl start {}'.format(service))
+
+
+def show_recovery_splash():
+    """
+    Shows the main recovery splash
+    """
+
+    _start_systemd_service('recovery-running-updater-splash')
+    import time
+    time.sleep(0.2)
+    _stop_systemd_service('recovery-boot-splash')
+
+
+def show_success_splash():
+    """
+    Shows the splash for a successful recovery
+
+    Returns:
+        int: Return code. Should be one of the members of the :class:`RC` class
+    """
+
+    _start_systemd_service('recovery-success-splash')
+    import time
+    time.sleep(0.2)
```

Comment:
would it be more practical to move these `0.2` as a variable above, along with MINUTE and HOUR?